compiler/src/test/scala/org/scalajs/core/compiler/test/JSExportTest.scala

```diff
@@ -423,24 +523,36 @@ class JSExportTest extends DirectTest with TestHelpers {
     class A {
       @JSExport
       object Nested
+
+      @JSExport
+      @ScalaJSDefined object Nested2 extends js.Object
     }
     """ hasErrors
     """
       |newSource1.scala:4: error: You may not export a nested object
       |      @JSExport
       |       ^
+      |newSource1.scala:7: error: You may not export a nested object
+      |      @JSExport
+      |       ^
     """
 
     """
     object A {
       @JSExport
       object Nested
+
+      @JSExport
+      @ScalaJSDefined object Nested2 extends js.Object
     }
     """ hasErrors
     """
       |newSource1.scala:4: error: You may not export a nested object
       |      @JSExport
       |       ^
+      |newSource1.scala:7: error: You may not export a nested object
+      |      @JSExport
+      |       ^
```

Comment:
This is currently OK, but we should revisit this. It does not make sense, that objects on Scala.js defined classes are exposed, but you cannot export objects on normal Scala classes.